tools/chasseturls.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+import re
+import click
+from datalad.api import Dataset
+
+
+@click.command()
+@click.argument("datasets", type=click.Path(file_okay=False, exists=True), nargs=-1)
+def main(datasets):
+    for ds in map(Dataset, datasets):
+        ds.repo.always_commit = False
+        changed = False
+        for path in ds.repo.get_annexed_files():
```

Comment:
sorry ,didn't spot right away.  As I have described in https://github.com/dandi/dandisets/issues/79#issue-980589807 -- we should operate not only on the files in the current tree, but all known to annex (e.g. in prior versions).  For that should operate at a "annex key" level and not on "per file" (in the tree).

Here is eg a command I ran to get a list of urls (after running this script).  You can see that we get some older one which has url not tuned:
```
dandi@drogon:/mnt/backup/dandi/dandisets$ python -c 'from pprint import pprint; from datalad.support.annexrepo import AnnexRepo; r = AnnexRepo("000029"); wis = (r.whereis(None, output="full", options=["--all"])); pprint({k: urls for k, v in wis.items() if (urls := v.get("00000000-0000-0000-0000-000000000001", {}).get("urls"))})'
{'SHA256E-s1024--a0955d0ce01d9251e092b8d1ff868f9a7b2b87ec4b23b74c3e9f3a2c0efcde02.dat': ['https://api.dandiarchive.org/api/dandisets/000029/versions/draft/assets/14c46a22-fe09-403b-b10f-4df1ed19e79e/download/',
                                                                                         'https://dandiarchive.s3.amazonaws.com/blobs/8d9/b3e/8d9b3ebf-d6df-495f-9391-71846e4c7e45?versionId=Nn50j4qP5_KDdnPdrc257E6WwTaqKach'],
 'SHA256E-s18792--1a765509384ea96b7b12136353d9c5b94f23d764ad0431e049197f7875eb352c.nwb': ['https://api.dandiarchive.org/api/assets/b3675aad-db07-4fd4-9cce-c95f1184e7a3/download/',
                                                                                          'https://dandiarchive.s3.amazonaws.com/blobs/2db/af0/2dbaf0fd-5003-4a0a-b4c0-bc8cdbdb3826?versionId=4BMLdtv04lfi7JDWFfUleHVzYazeFVL4'],
 'SHA256E-s357824--cc14bf369f827dddb3837d8d0f2596c806bcede09dd4a271ee0a399e103bccdd.nwb': ['https://dandiarchive.s3.amazonaws.com/girder-assetstore/96/86/9686486aac4244c8913f75d6c063e18d?versionId=Av6Ld7D8dsN7CIjGTrJafyPE8P2JafIY'],
 'SHA256E-s6393196--ebcdf64353eeb9b605533fdf8ee1c62a710b64e49593fe0f6de2e1ef8715cc7d.nwb': ['https://api.dandiarchive.org/api/assets/41d8709b-7079-4ca0-b9d1-34f382652e40/download/',
                                                                                            'https://dandiarchive.s3.amazonaws.com/blobs/e72/88a/e7288a2c-444c-42c5-b7bb-9a58c728992b?versionId=ewLlWfjwGUv9X9R0AYVv0KVMMKxLauh4'],
 'SHA256E-s6644036--d01c43576e3aeceed53a052defa8bbbf15b885146f518acf63d228cee44576ac.nwb': ['https://api.dandiarchive.org/api/assets/aef92c1b-bee4-41b3-8496-5e50ca370716/download/',
                                                                                            'https://dandiarchive.s3.amazonaws.com/blobs/180/9f5/1809f541-1cb1-48b8-b916-9a696bab488d?versionId=R6ghbdEuF4bXA9H4uVfqrwXnp8s8Eq1o'],
 'SHA256E-s7660100--130ee7635334db52b6e36e39d7c889d18b5fc43ae3a1298f1b12984399f26119.nwb': ['https://api.dandiarchive.org/api/assets/d6c659de-a62b-4ce9-ade6-15fbbdeda2d9/download/',
                                                                                            'https://dandiarchive.s3.amazonaws.com/blobs/c40/57c/c4057c5e-7af5-4370-878f-ccfc971aeba4?versionId=BNFHyDiPnx9nHU6TaBpqBk3bnMF1E8Xv']}
```

I guess you would need just to use `_call_annex` or just invoke it directly since those `rm_url` and `add_url_to_file` seems to not provide option to operate on keys (not paths)

Reply:
Fixed.